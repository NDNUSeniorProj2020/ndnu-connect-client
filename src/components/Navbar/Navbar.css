.navbar-container {
	background-color: #21285C;
	color: #ffffff;
}

.ndnu-logo {
	height: 60px; 
	float: left; 
	margin-left: 20px;
}

.tutor-menu-item {
	margin-left: 100px;
}

.avatar {
	color: #f56a00; 
	background-color: #fde3cf; 
	
}

.avatar-dropdown-container {
	float: right;
	margin: 10px 20px 0 0;
	cursor: pointer;
}

.dropdown-spacing {
	padding: 8px 35px 8px 35px;
}