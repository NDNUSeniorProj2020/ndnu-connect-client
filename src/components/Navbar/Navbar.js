import React from 'react';
import { withRouter } from 'react-router-dom';
import 'antd/dist/antd.css';
import { Avatar, Dropdown, Icon, Menu } from 'antd';

import logo from '../../assets/ndnu-logo.jpg';
import './Navbar.css';

const { SubMenu } = Menu;

export class ConnectedNavbar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      current: 'mail',
    };
  }

  handleClick = e => this.setState({ current: e.key });

  render() {
    const avatarMenu = (
      <Menu>
        <Menu.Item key="0" className="dropdown-spacing"><Icon type="setting" theme="filled" />Settings</Menu.Item>
        <Menu.Divider />
        <Menu.Item key="1" className="dropdown-spacing" onClick={this.props.handleLogout}>
          <Icon type="logout" />Logout
        </Menu.Item>
      </Menu>
    );

    return (
      <div id={'navbar'}>
        <a href="https://www.ndnu.edu"><img  src={logo} className="ndnu-logo" alt="NDNU Logo" /></a>

        <Dropdown overlay={avatarMenu} trigger={['click']} className="avatar-dropdown-container">
          <Avatar size="large" className="avatar">J</Avatar>
        </Dropdown>

        <Menu onClick={this.handleClick} selectedKeys={[this.state.current]} mode="horizontal" className="navbar-container">
          <SubMenu
            className="tutor-menu-item"
            title={
              <span className="submenu-title-wrapper">
                <Icon type="book" />
                Tutor
              </span>
            }
          >
            <Menu.Item key="setting:1">Student</Menu.Item>
            <Menu.Item key="setting:2">Tutor</Menu.Item>
          </SubMenu>
          <Menu.Item key="jobs" onClick={() => this.props.history.push('/jobs')}>
            Jobs | Internships
          </Menu.Item>
          <Menu.Item key="alumni">
            Alumni
          </Menu.Item>
          <Menu.Item key="forum">
            Open Forum
          </Menu.Item>
        </Menu>
      </div>
    );
  }
}

export default withRouter(ConnectedNavbar);
          