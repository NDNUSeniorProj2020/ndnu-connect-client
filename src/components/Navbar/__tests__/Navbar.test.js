import React from 'react';
import { shallow } from 'enzyme';

import { ConnectedNavbar } from '../Navbar';

describe('tests for Navbar', () => {
	describe('snapshot tests', () => {
		it('renders without crashing', () => {
			const tree = shallow(<ConnectedNavbar />);
			expect(tree).toMatchSnapshot();
		});
	});

	describe('integration tests', () => {
		it('should change current state when handleClick function is called', () => {
			const wrapper = shallow(<ConnectedNavbar />);

			wrapper.instance().handleClick({ key: 'jobs' });
			expect(wrapper.state('current')).toEqual('jobs');
		});
	});
});
