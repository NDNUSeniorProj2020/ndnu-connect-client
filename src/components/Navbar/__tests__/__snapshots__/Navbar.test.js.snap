// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Navbar snapshot tests renders without crashing 1`] = `ShallowWrapper {}`;
